modules/mono/csharp_script.cpp

```diff
@@ -2840,6 +2841,7 @@ void CSharpScript::initialize_for_managed_type(Ref<CSharpScript> p_script, GDMon
 
 	p_script->valid = true;
 	p_script->tool = p_script->script_class->has_attribute(CACHED_CLASS(ToolAttribute));
+	p_script->reload_invalidated = false;
```

Comment:
With the current changes this was wrong as it means it can be reloaded. We want the opposite. However with the changes I suggested (`reload_invalidated == true` means can reload) it should be fine as is.